src/samplers/statsd.c

```diff
@@ -48,28 +49,36 @@ static void statsd_run_recvmmsg(struct brubeck_statsd *statsd, int sock)
 			continue;
 		}
 
-		/* store stats */
-		brubeck_atomic_add(&server->stats.metrics, SIM_PACKETS);
-		brubeck_atomic_add(&statsd->sampler.inflow, SIM_PACKETS);
-
 		for (i = 0; i < SIM_PACKETS; ++i) {
-			char *buf = msgs[i].msg_hdr.msg_iov->iov_base;
-			int len = msgs[i].msg_len;
-
-			if (brubeck_statsd_msg_parse(&msg, buf, len) < 0) {
-				if (msg.key_len > 0)
-					buf[msg.key_len] = ':';
-
-				log_splunk("sampler=statsd event=bad_key key='%.*s'", len, buf);
-
-				brubeck_server_mark_dropped(server);
-				continue;
+			char *end = msgs[i].msg_hdr.msg_iov->iov_base + msgs[i].msg_len;
+			*end = '\0';
+
+			char *tok;
+			char *saveptr;
+			tok = strtok_r(msgs[i].msg_hdr.msg_iov->iov_base, "\n", &saveptr);
```

Comment:
This can be accomplished much easier with a `memchr` call, since `strtok` is an anypattern (even in the reentrant version), and you need to call `strlen` anyway on the buffer.